src/hooks/useCurrentExecutionStatus.ts

```diff
@@ -0,0 +1,142 @@
+      const latestReceivedNonceCall = await targetApi.rpc.state.call<Codec>(
+        latestReceivedNonceMethodName,
+        laneId,
+        blockHash.toJSON()
+      );
+
+      // @ts-ignore
+      const latestReceivedNonceCallType = targetApi.registry.createType('MessageNonce', latestReceivedNonceCall);
+      const latestReceivedNonce = latestReceivedNonceCallType.toString();
+      setNonceOfTargetFinalizedBlock(parseInt(latestReceivedNonce));
+    };
+
+    const getLatestReceivedNonce = async () => {
+      const latestReceivedNonceCall = await targetApi.rpc.state.call<Codec>(latestReceivedNonceMethodName, laneId);
+      // @ts-ignore
+      const latestReceivedNonceCallType = targetApi.registry.createType('MessageNonce', latestReceivedNonceCall);
+      setLatestReceivedNonceRuntimeApi(parseInt(latestReceivedNonceCallType.toString()));
+    };
+
+    getNonceByHash();
+    getLatestReceivedNonce();
+  }, [
+    areApiLoading,
+    bestBlockFinalizedOnTarget,
+    currentTransaction,
+    laneId,
+    latestReceivedNonceMethodName,
+    targetApi.registry,
+    targetApi.rpc.chain,
+    targetApi.rpc.state
+  ]);
+
+  useEffect(() => {
+    if (!areApiLoading || !currentTransaction) {
+      return;
+    }
+
+    const stepEvaluator = (transactionValue: string | number | null, chainValue: number | null) => {
+      if (!transactionValue || !chainValue) {
+        return false;
+      }
+      return chainValue > transactionValue;
+    };
+
+    const isDone = (status: boolean | boolean) => (currentTransaction.block && status ? 'DONE' : 'RUNNING');
+    const step1 = currentTransaction.block ? `included in block ${currentTransaction.block}` : isDone(false);
+    const step2 = stepEvaluator(currentTransaction.block, parseInt(bestBlockFinalized));
+    const step3 = stepEvaluator(currentTransaction.block, parseInt(importedHeaders));
+    const step4 = stepEvaluator(currentTransaction.messageNonce, latestReceivedNonceRuntimeApi);
+    const step5 = stepEvaluator(currentTransaction.messageNonce, nonceOfTargetFinalizedBlock);
+    const step6 = stepEvaluator(currentTransaction.messageNonce, latestReceivedNonceOnSource);
+
+    const steps = [
+      { chainType: sourceChain, label: 'Including message in block', status: step1 },
+      { chainType: sourceChain, label: 'Waiting for block finality on source', status: isDone(step2) },
+      { chainType: targetChain, label: 'Source Chain block finality received on target', status: isDone(step3) },
+      { chainType: targetChain, label: 'Message delivered', status: isDone(step4) },
+      { chainType: targetChain, label: 'Message finality', status: isDone(step5) },
+      { chainType: sourceChain, label: 'Delivery confirmation on source', status: isDone(step6) }
+    ];
+
+    if (step6) {
+      dispatchTransaction(
+        TransactionActionCreators.updateTransactionStatus({ status: TransactionStatusEnum.COMPLETED })
+      );
+    }
```

Comment:
Hook dispatching a transaction looks a bit weird to me, let me clarify what I mean.

Since you have split hooks and global state imho they should be independent from each other. However here you are introducing a tight coupling between the two. If I create a component that is only interested in transaction status (i.e. doesn't call `useCurrentExecutionStatus`) I'll never see the transaction being completed, which is not quite what I expect.

That makes me feel this is not the right abstraction you have here and having such important logic within a not-quite-related hook is not the right design choice. I think that status tracking should be isolated to one hook (and actually as mentioned some place else we should be able to track multiple transactions at once, but that can be done in the future).